### ISSUE TITLE
[FEATURE REQUEST] Command to build selected module with all required modules

### ISSUE THREAD

**Issue by szymex** [krasa/MavenHelper#7]
Command to build selected module with all required modules. 
For example:

```
mvn install -pl module-abc -am
```

**Comment by jeacott1** [comment]
this is definitely required - its currently impossible to run individual tests in multi module projects with this as-is.
this plugin also seems to completely ignore any use or 'resolve-workspace-artifacts'.

**Comment by krasa** [comment]
Is $moduleName$ all that is needed? 
If so, then you can try it:
[MavenRunHelper.zip](https://github.com/krasa/MavenHelper/files/8176193/MavenRunHelper.zip)

**Comment by jeacott1** [comment]
@krasa actually had a brief look at the code - I think all thats really needed is in the test actions,  hunt for the ''resolve-workspace-artifacts' boolean setting, and if it exists add -am to the maven cmd options.
if there was a place to do it in intellij I'd suggest a means to properly customise the maven options, but for my purposes all I actually want is to set the root folder (module), and pass -am

**Comment by jeacott1** [comment]
@krasa  - tried the file you provided above - seems broken now.  - in the cmd generated it now says -Dtest=NOT_RESOLVED :(

**Comment by krasa** [comment]
Fixed:
[MavenRunHelper.zip](https://github.com/krasa/MavenHelper/files/8176351/MavenRunHelper.zip)

resolve-workspace-artifacts settings is here, it should now work for tests too:
![image](https://user-images.githubusercontent.com/1160875/156530948-b7d033b0-854d-4b35-a4f7-96bc158847bc.png)

**Comment by jeacott1** [comment]
@krasa thanks, better!
there might be a bug in that whenever I edit the settings I need to restart the ide or the plugin context menus are completely gone.

ok, I got what I wanted working - but I also needed to add -f /path/to/my/aggregator-root/weirdnamedpom.xml
thank you! - I don't much like intellij but this makes dealing with it soooo much better,

the only thing I couldn't find was a variable or some means to find the project root folder or a specific module folder - the line above finds the absolute path, which works.  is there a way?

**Comment by krasa** [comment]
"there might be a bug in that whenever I edit the settings I need to restart the ide or the plugin context menus are completely gone."

That sounds super strange, is there something in the log?


So you want a macro for a maven root project folder? That should be no problem.

**Comment by jeacott1** [comment]
@krasa I'll get back to you on that - havent had it happen again in a while.
fyi my previous settings didn't quite work.   
I tried adding a -Dmaven.multiModuleProjectDirectory=/path/to/my/aggregator-root  but unfortunately it doesn't override the default one so I cant test other modules except the root module.
I also added -f weirdaggpom.xml -Dit.test=$classWithMethod$
$classWithMethod$ is only giving me the classname. I had hoped it might generate className#testMethodName  - should it?

fyi, in my case my parent pom is strictly an aggregator. there is no parent set to this pom in any of the submodules.

**Comment by krasa** [comment]
$classWithMethod$  requires a caret in the method.

**Comment by jeacott1** [comment]
@krasa yep - cursor/caret in the method - no method in result

![image](https://user-images.githubusercontent.com/16600100/156543292-a6e130d8-263f-4e0f-916a-b4ee0802e9e6.png)
![image](https://user-images.githubusercontent.com/16600100/156543386-e73b3d90-4991-4c74-b397-4110e136bad3.png)

I will note that intellij cant resolve various dependencies in this project (maven can, maven in intellij can, and in this particular module (same folder as the root pom - maven helper can build and run this test - it just doesnt give me the #test)

**Comment by krasa** [comment]
Is the method executable by IntelliJ itself?
![image](https://user-images.githubusercontent.com/1160875/156544106-9bd7e001-f3eb-4a0f-8c9e-2d0e61e4d071.png)

**Comment by jeacott1** [comment]
@krasa no, and I have no idea how to make it so. I guess thats the reason.

**Comment by krasa** [comment]
It needs a JUnit dependency and the annotation, or some alternative.
